### ISSUE TITLE
rerender breaks bind mount

### ISSUE THREAD

**Issue by fredwangwang** [hashicorp/consul-template#1436]
### Consul Template version

latest

### Expected behavior

when consul-template rerenders the file at location A, the file at location B which is a bind mount of A **does not** get updated.

### Actual behavior

when consul-template rerenders the file at location A, the file at location B which is a bind mount of A **should** get updated.

### Steps to reproduce
this is basically what happens behind scene

```bash
# has to run on linux machine
touch a b # create the source and mount point
echo 1 > a
sudo mount --bind a b # bind mount a --> b
cat a
cat b
rm a # break the bind, which is essentially "AtomicWrite" is doing
echo 2 > a
cat a # 2
cat b # 1

sudo umount b # clean up
rm a b
```

so basically the line breaks bind is: https://github.com/hashicorp/consul-template/blob/91c3dd4189ad3d8d21130694f436e9706e3ccf5a/renderer/renderer.go#L178

### References

https://stackoverflow.com/a/53548078/10634687

**Comment by eikenb** [comment]
Hey @fredwangwang, sorry you're having a problem.

The standard way to update a file atomically is to write to a temp file (non-atomic) then rename/move that file over the old one (atomic). You can't just write to the existing file in place w/o the risk of a partial file write. 

So I'm not really sure what can be done here with consul-template as this is pretty much the correct way to do this. You should be able to get it working if you create the bind mount to a directory instead of a file, as it uses the same directory for everything.

Otherwise I'm open to ideas.

**Comment by fredwangwang** [comment]
Hi @eikenb, thanks for your reply!

It does look like there is no reliable way to atomicly update the file while preserving the inode, oh well..

> create the bind mount to a directory instead of a file

This is exactly where we landed! Thanks again for your inputs on this.
